### PR TITLE
python-gyp: update with python3 support + drop python2 dependency in nss build

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
Follow-up (of sorts) to #91746

This probably doesn't fix the darwin build, but at least it drops the python2 dependency.
Should have checked this sooner…

cc @vcunat @flokli 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
Well, why not try a darwin build anywa; deps should be in cache ATM.
@GrahamcOfBorg build nss

**Comment by ajs124** [comment]
The missing thing always seems to be libtool, so we can just try and give it libtool and see what that does.

**Comment by vcunat** [comment]
Oh, I missed that.  OK.
@GrahamcOfBorg build nss

**Comment by vcunat** [comment]
By a quick search on the internet, I suspect we want libtool from `darwin.cctools` package.  (I'm probably not allowed to push into _this_ branch.)

**Comment by flokli** [comment]
@GrahamcOfBorg build nss

**Comment by flokli** [comment]
@GrahamcOfBorg build nss

**Comment by ajs124** [comment]
cc @NixOS/darwin-maintainers this times out on ofborg

**Comment by vcunat** [comment]
Timed out at `[1217/1220]`... so that libtool error seems fixed for sure and even overall I think there's good chance it will be OK.

**Comment by flokli** [comment]
I tried reproduce this on a borrowed Mac,

For some reason, I ran into some  `No such file or directory: xcodebuild` errors, which don't occur on ofborg. I was able to migitate the by also adding some of the frameworks and `xcbuild`, but I don't feel qualified to decide on whether this is necessary or not, given it seems to not occur on ofborg. I defer to @LnL7 on that.

**Comment by vcunat** [comment]
Could that have been by you not using sandboxing?

**Comment by flokli** [comment]
> Could that have been due to you not using sandboxing?

With sandboxing enabled, the sandbox complained trying to invoke `xcodebuild`, without it, MacOS complained about xcode not being installed properly.

I don't really know - I hope if ofborg builds, hydra will do so aswell. I really am not familiar enough with this architecture to judge.

**Review by LnL7 — APPROVED** [review]
The darwin build succeeded on 2 different hosts.

@flokli not sure what caused your issue, but at least it shouldn't happen on hydra from what I can tell

**Comment by vcunat** [comment]
Confirmed on Hydra: https://hydra.nixos.org/build/124199611

**Pull request merged**
